### PR TITLE
Csh issue 1118 node 10.15.3

### PR DESCRIPTION
**Pull request by mans0954**
### Description

Updates real-time to run on Node 10

#### Related Issues / PRs

Related:

https://github.com/overleaf/dev-environment/pull/217
https://github.com/overleaf/chef-config/pull/343

Closes: https://github.com/overleaf/real-time/issues/48

Supersedes: #32 

#### Potential Impact

Node 6 to Node 10 significant upgrade by 2 Node LTS versions

`flashsocket` removed from `socket.io` transports

#### Manual Testing Performed

- [ ]
- [ ]

#### Deployment Checklist

Issue:

https://github.com/overleaf/real-time/issues/48

Pull Requests:

https://github.com/overleaf/real-time/pull/49
https://github.com/overleaf/chef-config/pull/343
https://github.com/overleaf/dev-environment/pull/217

Plan:

Get buildno from http://builds.sharelatex.com/job/real-time-sharelatex/job/csh-issue-1118-node-10.15.3/
Get LIVE and DARK from `knife data bag show server_groups staging --format=json | jq '."real-time"'`

```
cd ~/src/sde/chef
git checkout csh-issue-1118-node-10.15.3-rt
knife cookbook upload real_time
rake deploy:real_time_dark[staging,csh-issue-1118-node-10.15.3,buildno]
```
Test on https://dark.overleaf.staging/ whilst tailing /var/www/real-time-{DARK}-sharelatex/shared/log/production.log on sharelatex@sl-lin-stag-web-n where n is indicated by the `SERVERID` cookie. Look for level":50s and any thrown exceptions. Look for the projectid to confirm looking at right logs.

If all looks good:
```
rake server_groups:flip[staging,real-time]
rake deploy:lb[staging]
```
Test again on https://www.overleaf.staging/

If still good:

Merge the real-time `csh-issue-1118-node-10.15.3` branch into master https://github.com/overleaf/real-time/pull/49

Get the build number from http://builds.sharelatex.com/job/real-time-sharelatex/job/master/
```
rake deploy:real_time_dark[staging,master,buildno]
```
Test on https://dark.overleaf.staging/ whilst tailing /var/www/real-time-{DARK}-sharelatex/shared/log/production.log on sharelatex@sl-lin-stag-web-n where n is indicated by the `SERVERID` cookie

If all looks good:
```
rake server_groups:flip[staging,real-time]
rake deploy:lb[staging]
```
Test again on https://www.overleaf.staging/

If still good, deploy to production:
```
rake deploy:real_time_dark[production,master,buildno]
```
Test on https://dark.overleaf.com

If all looks good:
```
rake server_groups:flip[production,real-time]
rake deploy:lb[production]
```
Test on https://www.overleaf.com

merge chef-config ` csh-issue-1118-node-10.15.3-rt` branch into master https://github.com/overleaf/chef-config/pull/343

Merge dev-environment `csh-issue-1118-node-10.15.3-rt` branch into master https://github.com/overleaf/dev-environment/pull/217

### CODE REVIEW

File: app.coffee [hunk 1]
```diff
@@ -45,7 +45,7 @@ io.configure ->
 	# gzip uses a Node 0.8.x method of calling the gzip program which
 	# doesn't work with 0.6.x
 	#io.enable('browser client gzip')
-	io.set('transports', ['websocket', 'flashsocket', 'htmlfile', 'xhr-polling', 'jsonp-polling'])
```
Comment (henryoswald) [inline app.coffee]: I think it is fine to loose `flashsocket`, @briangough I don't think we see much of them, agree?

**Comment by henryoswald** [comment]
This pr contains commits from a non OL employee, we need to check they have signed us the IP, simpler would be to remove their commits.

**Comment by mans0954** [comment]
> This pr contains commits from a non OL employee, we need to check they have signed us the IP, simpler would be to remove their commits.

I've already checked that with @jdleesmiller - the third party has signed the IP.

File: app.coffee (hunk 1 above)
Comment (briangough) [reply app.coffee]: Yes we should remove that, we don't need to support flash these days.

**Pull request merged**